src/wp-includes/block-template-utils.php

```diff
@@ -1012,3 +1014,65 @@ function wp_generate_block_templates_export_file() {
 
 	return $filename;
 }
+
+/**
+ * Helper function to get the Template Hierarchy for a given slug.
+ * We need to Handle special cases here like `front-page`, `singular` and `archive` templates.
+ *
+ * Noting that we always add `index` as the last fallback template.
+ *
+ * @since 6.1.0
+ *
+ * @param string  $slug            The template slug to be created.
+ * @param boolean $is_custom       Indicates if a template is custom or part of the template hierarchy.
+ * @param string  $template_prefix The template prefix for the created template. This is used to extract the main template type ex. in `taxonomy-books` we extract the `taxonomy`.
+ *
+ * @return array<string> The template hierarchy.
+ */
+function get_template_hierarchy( $slug, $is_custom = false, $template_prefix = '' ) {
```

Comment:
Shouldn't this be prefixed with `wp_`? Why? 

- consistency
- to avoid any naming conflicts in the wild

Reply:
Oh I see, the other function names in that same file do not use the `wp_` prefix. Interesting. Okay, seems it's consistent within its context.